homework2/hw1.py

```diff
@@ -0,0 +1,68 @@
+"""
+Given a file containing text. Complete using only default collections:
+    1) Find 10 longest words consisting from largest amount of unique symbols
+    2) Find rarest symbol for document
+    3) Count every punctuation char
+    4) Count every non ascii char
+    5) Find most common non ascii char for document
+"""
+import string
+from collections import Counter
+from typing import List
+
+
+def get_longest_diverse_words(file_path: str) -> List[str]:
+    with open(file_path) as file:
+        lines = []
+        for line in file:
+            lines += line.strip().split()
+
+    return sorted(lines, key=lambda x: len(x) and len(Counter(x)), reverse=True)[:10]
+
+
+def get_rarest_char(file_path: str) -> str:
+    with open(file_path) as file:
+        chars: dict = Counter("".join(file))
+        return min(chars.items(), key=(lambda key: chars[key]))[0]
```

Comment:
This function returns wrong answer. For example, `C` has count of 22.

Reply:
You are correct, capital _"C"_ occurs 22 times, but lowercase - 5497, should I consider the case of letters?

Reply:
You can use `Counter.most_common` to get sorted list of items here.

Reply:
Remove it to `chars.most_common()[-1][0]`